### PR TITLE
decompile unk_02005CFC.s into sound_chatot.c

### PR DESCRIPTION
**Pull request by remicalixte**

### CODE REVIEW

File: arm9/src/unk_020040F4.c [hunk 1]
```diff
@@ -650,14 +648,13 @@ THUMB_FUNC u16 FUN_02004900(u32 param0)
...
 }
 
-THUMB_FUNC u32 GF_MIC_StartAutoSampling(u32 param0)
+THUMB_FUNC u32 GF_MIC_StartAutoSampling(struct MIC_SamplingData *param0)
 {
     return MIC_StartAutoSampling(param0);
 }
 
-THUMB_FUNC u32 GF_MIC_StopAutoSampling(u32 param0)
+THUMB_FUNC u32 GF_MIC_StopAutoSampling()
```
Comment (PikalaxALT) [inline arm9/src/unk_020040F4.c:656]: Functions that take no args, should be explicitly declared with arg list `void`

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: arm9/lib/NitroSDK/include/SPI_mic.h [hunk 2]
```diff
@@ -1,6 +1,23 @@
 #ifndef NITRO_SPI_MIC_H_
 #define NITRO_SPI_MIC_H_
 
+#include "global.h"
```
Comment (red031000) [inline arm9/lib/NitroSDK/include/SPI_mic.h:4]: including global.h in NitroSDK can cause some weirdness with circular includes, I would generally discourage it

File: arm9/src/sound_chatot.c [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    }
+    else if (r0 >= 30 && r0 < 128)
+    {
+        return 2;
+    }
+
+    return 0;
+}
+
+THUMB_FUNC u32 FUN_02005F14(s32 param0)
```
Comment (red031000) [inline arm9/src/sound_chatot.c:168]: can this work with a BOOL?

File: arm9/src/sound_chatot.c [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    }
+    st0.unk0c = 0x4174;
+    st0.unk10 = 0;
+    st0.unk14 = 0;
+    st0.unk18 = 0;
+
+    return GF_MIC_StartAutoSampling(&st0);
+}
+
+THUMB_FUNC void Chatot_stopRecording()
```
Comment (red031000) [inline arm9/src/sound_chatot.c:113]: void in parameter list

File: arm9/src/sound_chatot.c [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    if (*r5 == 1)
+    {
+        FUN_02004C3C(14);
+        FUN_02004A04(14);
+    }
+
+    *r4 = 0;
+}
+
+THUMB_FUNC u32 Chatot_startRecording()
```
Comment (red031000) [inline arm9/src/sound_chatot.c:94]: void in parameter list

File: arm9/src/sound_chatot.c [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+    u32 res = FUN_02004A6C(&st8, 14);
+    FUN_02004AF8(14, param2);
+    *st4 = 1;
+    FUN_02005E80(0);
+
+    return res;
+}
+
+THUMB_FUNC void FUN_02005DFC()
```
Comment (red031000) [inline arm9/src/sound_chatot.c:80]: void in parameter list

File: arm9/src/sound_chatot.c [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+#include "sound_chatot.h"
+#include "SPI_mic.h"
+#include "math_util.h"
+#include "proto.h"
+#include "sav_chatot.h"
+#include "sound.h"
+#include "unk_020040F4.h"
+#include "unk_020051F4.h"
+
+THUMB_FUNC BOOL FUN_02005CFC()
```
Comment (red031000) [inline arm9/src/sound_chatot.c:11]: void in parameter list

File: arm9/lib/NitroSDK/include/SPI_mic.h [hunk 8]
```diff
@@ -1,6 +1,23 @@
...
+    u32 unk08;
+    u32 unk0c;
+    u32 unk10;
+    u32 unk14;
+    u32 unk18;
+};
+
 void MIC_Init(void);
+u32 MIC_StartAutoSampling(struct MIC_SamplingData *data);
+u32 MIC_StopAutoSampling();
```
Comment (red031000) [inline arm9/lib/NitroSDK/include/SPI_mic.h:20]: void in parameter list

File: include/sound_chatot.h [hunk 9]
```diff
@@ -0,0 +1,19 @@
+#ifndef POKEDIAMOND_SOUND_CHATOT_H
+#define POKEDIAMOND_SOUND_CHATOT_H
+
+#include "global.h"
+#include "sav_chatot.h"
+
+BOOL FUN_02005CFC();
```
Comment (red031000) [inline include/sound_chatot.h:7]: void in parameter list for all these

File: include/unk_020040F4.h [hunk 10]
```diff
@@ -65,8 +66,8 @@ u8 FUN_020048D0(u16 param0);
 u32 GF_SndPlayerGetSeqNo(u32 *param0);
 u32 FUN_020048F4(u16 param0);
 u16 FUN_02004900(u32 param0);
-u32 GF_MIC_StartAutoSampling(u32 param0);
-u32 GF_MIC_StopAutoSampling(u32 param0);
+u32 GF_MIC_StartAutoSampling(struct MIC_SamplingData *param0);
+u32 GF_MIC_StopAutoSampling();
```
Comment (red031000) [inline include/unk_020040F4.h:70]: void in parameter list (u32? this is a bit confusing for me)

**Review by red031000 — CHANGES_REQUESTED** [review]

File: arm9/src/sound_chatot.c (hunk 3 above)
Comment (remicalixte) [reply arm9/src/sound_chatot.c]: yes

File: include/unk_020040F4.h (hunk 10 above)
Comment (remicalixte) [reply include/unk_020040F4.h]: I removed the u32 to replace it with a struct. Or are you talking about the return type?

Comment (red031000) [reply include/unk_020040F4.h]: no I mean, in GF_MIC_StopAutoSampling you replaced the u32 with a blank parameter list, so I'm unsure what should be there, void or a u32, or something else

File: arm9/lib/NitroSDK/include/SPI_mic.h [hunk 11]
```diff
@@ -1,6 +1,20 @@
...
+    int unk08;
+    int unk0c;
+    int unk10;
+    int unk14;
+    int unk18;
+};
+
 void MIC_Init(void);
+int MIC_StartAutoSampling(struct MIC_SamplingData *data);
+int MIC_StopAutoSampling();
```
Comment (red031000) [inline arm9/lib/NitroSDK/include/SPI_mic.h:17]: void here, or appropriate parameters

**Review by red031000 — CHANGES_REQUESTED** [review]

File: include/unk_020040F4.h (hunk 10 above)
Comment (remicalixte) [reply include/unk_020040F4.h]: This was a mistake I did when decompiling `GF_MIC_StopAutoSampling`. the argument wasn't even used and now that I decompiled code calling it, it's clear it just doesn't have any argument.

**Review by red031000 — APPROVED** [review]

**Review by ScottNortonPhD — APPROVED** [review]

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**